airbyte-integrations/connectors/destination-bigquery-denormalized/Dockerfile

```diff
@@ -17,5 +17,5 @@ ENV ENABLE_SENTRY true
 
 COPY --from=build /airbyte /airbyte
 
-LABEL io.airbyte.version=1.1.11
+LABEL io.airbyte.version=2.0.0
```

Comment:
why major version change?

Reply:
There is no backward compatibility with existing connectors. The structure of the destination will be different now.
I decided that by the version, I could highlight it. 
What would you recommend doing in this case?

Reply:
What will happen if a user upgrades to this version of connector? Will their jobs stop working until they do a full reset?

Reply:
There are three possible cases: 
1. Everything works as was before.
2. Sync skip part of the data
3. Sync fails

Case #1 is possible if you don't have any arrays. But usually, customers use BigQuery Denormalized destinations when they have complex types like arrays or records, and the usual destination can't normalize them. The result is entirely dependent on the user connector schema.

I haven't faced a connector update without backward compatibility. Shall we implement a feature that will force you to reset a destination if you upgraded the version without backward compatibility?

Reply:
The solution here is certainly not to force a reset of a destination after the upgrade - that's the last resort. We cannot release a new version of BQ Denormalized destination connector if the upgrade requires a reset, neither if the upgrade results in parts of data being skipped that were not skipped before. There needs to be better handling of the upgrade scenario here.

Reply:
before we can ship this, I need to understand the scope of disruption that this change will cause to existing customers and possible ways to avoid the disruption. As a starting point, please identify and describe the conditions that result in each of these behaviors post-upgrade and codify them as integration tests

1. Everything works as was before.
2. Sync skips part of the data
3. Sync fails

Once we have these conditions codified we'll have to look for solutions to minimize disruptions.

Reply:
#14440 investigation issue